core/shared/src/main/scala/internal/JsonLogAppender.scala

```diff
@@ -0,0 +1,59 @@
+package zio.logging.internal
+
+import zio.Cause
+import zio.logging.JsonEscape
+
+private[logging] class JsonLogAppender extends LogAppender {
```

Comment:
Due to recursion it would probably be necessary for `writingValueOfKey` to not be expressed as a boolean, but rather, to be expressed using some sort of stack or counters.